Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -1331,8 +1331,13 @@ void RenderBlock::paintObject(PaintInfo& paintInfo, const LayoutPoint& paintOffs
         paintFloats(paintInfo, scrolledOffset, paintPhase == PaintPhase::Selection || paintPhase == PaintPhase::TextClip || paintPhase == PaintPhase::EventRegion);
 
     // 5. paint outline.
-    if ((paintPhase == PaintPhase::Outline || paintPhase == PaintPhase::SelfOutline) && hasOutline() && style().visibility() == Visibility::Visible)
+    if ((paintPhase == PaintPhase::Outline || paintPhase == PaintPhase::SelfOutline) && hasOutline() && style().visibility() == Visibility::Visible) {
+        // Don't paint focus ring for anonymous block continuation because the
+        // inline element having outline-style:auto paints the whole focus ring.
+        bool hadNoOutlineAuto = style().outlineStyleIsAuto() == OutlineIsAuto::On;
```

Comment:
not sure if the variable name reflects style().outlineStyleIsAuto() == OutlineIsAuto::On;

Reply:
I updated to above since your point is fair, it was unclear and didn't reflect what this is doing.